src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -2,68 +2,70 @@
+    private static final String INSERT_BETWEEN_VALUES = SPACE_DELIMITER + MINUS_DELIMITER
+            + SPACE_DELIMITER;
 
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-
         DateTimeFormatter formatter = DateTimeFormatter.ofPattern(DATE_FORMAT);
         LocalDate startOfPeriod = LocalDate.parse(dateFrom, formatter);
         LocalDate endOfPeriod = LocalDate.parse(dateTo, formatter);
-
         if (!startOfPeriod.equals(endOfPeriod)) {
             endOfPeriod = endOfPeriod.plusDays(1);
         }
-
-        List<Employee> listEmployee = new ArrayList<>();
-
-        for (String name : names) {
-            listEmployee.add(new Employee(name, 0));
+        String[] report = new String[names.length];
+        StringBuilder stringBuilder = new StringBuilder();
+        for (int i = 0; i < names.length; i++) {
+            report[i] = stringBuilder.append(names[i]).append(INSERT_BETWEEN_VALUES)
+                                     .append(ZERO_VALUE).toString();
+            stringBuilder.setLength(ZERO_VALUE);
         }
-
-        List<DataOfEmployees> dataEmployees = new ArrayList<>();
-
-        for (String dataString : data) {
-            String[] fieldsOfDataEmployee = dataString.split(SPACE_DELIMITER);
-            LocalDate date = LocalDate.parse(fieldsOfDataEmployee[0], formatter);
-            String employeeName = fieldsOfDataEmployee[1];
-            int workedHours = Integer.parseInt(fieldsOfDataEmployee[2]);
-            int salaryRate = Integer.parseInt(fieldsOfDataEmployee[3]);
-            dataEmployees.add(new DataOfEmployees(date, employeeName, workedHours, salaryRate));
-        }
-
-        for (DataOfEmployees dataEmployee : dataEmployees) {
-            if (dataEmployee.getDate().isAfter(startOfPeriod)
-                    && dataEmployee.getDate().isBefore(endOfPeriod)) {
-                for (Employee employee : listEmployee) {
-                    if (employee.getName().equals(dataEmployee.getName())) {
-                        int newSum = employee.getSalaryFromTo()
-                                + (dataEmployee.getWorkedHours() * dataEmployee.getSalaryRate());
-                        employee.setSalaryFromTo(newSum);
+        for (int i = 0; i < names.length; i++) {
+            for (String line : data) {
+                String[] fieldsOfData = line.split(SPACE_DELIMITER);
+                LocalDate date =
+                        LocalDate.parse(fieldsOfData[DATE_IN_DATA_POSITION], formatter);
+                if (line.contains(names[i]) && (date.isAfter(startOfPeriod)
+                        && date.isBefore(endOfPeriod))) {
+                    int workedHours = Integer.parseInt(fieldsOfData[WORKED_HOURS_POSITION]);
+                    int salaryRate =
+                            Integer.parseInt(fieldsOfData[SALARY_RATE_PER_HOUR_POSITION]);
+                    for (int j = 0; j < report.length; j++) {
```

Comment:
you don't need this loop. you can process data with two loops: for each name, for each line.

Reply:
I agree. Accepted.